sci-geosciences/qgis/qgis-2.18.2-r3.ebuild

```diff
@@ -0,0 +1,198 @@
+	dev-qt/qtcore:4
+	dev-qt/qtgui:4
+	dev-qt/qtscript:4
+	dev-qt/qtsvg:4
+	dev-qt/qtsql:4
+	dev-qt/qtwebkit:4
+	x11-libs/qscintilla:=[qt4(-)]
+	|| (
+		( || ( <x11-libs/qwt-6.1.2:6[svg] >=x11-libs/qwt-6.1.2:6[svg,qt4] ) >=x11-libs/qwtpolar-1 )
+		( x11-libs/qwt:5[svg] <x11-libs/qwtpolar-1 )
+	)
+	georeferencer? ( sci-libs/gsl:= )
+	grass? ( || ( >=sci-geosciences/grass-7.0.0:= ) )
+	mapserver? ( dev-libs/fcgi )
+	oracle? ( dev-db/oracle-instantclient:= )
+	postgres? ( dev-db/postgresql:= )
+	python? (
+		dev-python/PyQt4[X,sql,svg,webkit,${PYTHON_USEDEP}]
+		<dev-python/sip-4.19:=[${PYTHON_USEDEP}]
+		dev-python/qscintilla-python[${PYTHON_USEDEP}]
+		dev-python/python-dateutil[${PYTHON_USEDEP}]
+		dev-python/httplib2[${PYTHON_USEDEP}]
+		dev-python/jinja[${PYTHON_USEDEP}]
+		dev-python/markupsafe[${PYTHON_USEDEP}]
+		dev-python/pygments[${PYTHON_USEDEP}]
+		dev-python/pytz[${PYTHON_USEDEP}]
+		dev-python/requests[${PYTHON_USEDEP}]
+		dev-python/six[${PYTHON_USEDEP}]
+		postgres? ( dev-python/psycopg:2[${PYTHON_USEDEP}] )
+		${PYTHON_DEPS}
+	)
+	dev-db/sqlite:3
+	>=dev-db/spatialite-4.1.0
+	app-crypt/qca:2[qt4,ssl]
+"
+
+DEPEND="${RDEPEND}
+	sys-devel/bison
+	sys-devel/flex"
+
+DOCS=( BUGS ChangeLog NEWS )
+
+PATCHES=( "${FILESDIR}"/${P}-desktop-icon.patch )
+
+# Disabling test suite because upstream disallow running from install path
+RESTRICT="test"
+
+pkg_setup() {
+	python-single-r1_pkg_setup
+}
+
+src_prepare() {
+	default
+
+	cd src/plugins || die
+	use georeferencer || cmake_comment_add_subdirectory "georeferencer"
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DQGIS_MANUAL_SUBDIR=/share/man/
+		-DBUILD_SHARED_LIBS=ON
+		-DQGIS_LIB_SUBDIR=$(get_libdir)
+		-DQGIS_PLUGIN_SUBDIR=$(get_libdir)/qgis
+		-DWITH_INTERNAL_DATEUTIL=OFF
+		-DWITH_INTERNAL_HTTPLIB2=OFF
+		-DWITH_INTERNAL_JINJA2=OFF
+		-DWITH_INTERNAL_MARKUPSAFE=OFF
+		-DWITH_INTERNAL_PYGMENTS=OFF
+		-DWITH_INTERNAL_PYTZ=OFF
+		-DWITH_INTERNAL_QWTPOLAR=OFF
+		-DWITH_INTERNAL_SIX=OFF
+		-DPEDANTIC=OFF
+		-DWITH_APIDOC=OFF
+		-DWITH_QSPATIALITE=ON
+		-DENABLE_TESTS=OFF
+		-DWITH_BINDINGS="$(usex python)"
+		-DWITH_GRASS7="$(usex grass)"
+		$(usex grass "-DGRASS_PREFIX=/usr/" "")
+		-DWITH_ORACLE="$(usex oracle)"
+		-DWITH_POSTGRESQL="$(usex postgres)"
+		-DWITH_PYSPATIALITE="$(usex python)"
+		-DWITH_SERVER="$(usex mapserver)"
+	)
+
+	if has_version '>=x11-libs/qwtpolar-1' &&  has_version 'x11-libs/qwt:5' ; then
+		elog "Both >=x11-libs/qwtpolar-1 and x11-libs/qwt:5 installed. Force build with qwt6"
+		if has_version '>=x11-libs/qwt-6.1.2' ; then
+			mycmakeargs+=(
+				"-DQWT_INCLUDE_DIR=/usr/include/qwt6"
+				"-DQWT_LIBRARY=/usr/$(get_libdir)/libqwt6-qt4.so"
+			)
+		else
+			mycmakeargs+=(
+				"-DQWT_INCLUDE_DIR=/usr/include/qwt6"
+				"-DQWT_LIBRARY=/usr/$(get_libdir)/libqwt6.so"
+			)
+		fi
+	fi
+
+	cmake-utils_src_configure
+}
+
+src_install() {
+	cmake-utils_src_install
+
+	domenu debian/qgis.desktop
+
+	local types="qgs qml qlr qpt"
+	local sizes="16 22 24 32 48 96 128 256"
```

Comment:
I quite like having them all obvious up front, but agree it is wasteful. Removed and made loop counters local.